dynamic_instrumentation/src/cast.rs

```diff
@@ -0,0 +1,117 @@
+use rustc_index::vec::IndexVec;
+use rustc_middle::{
+    mir::{
+        CastKind, Local, LocalDecl, Mutability, Operand, ProjectionElem, Rvalue, SourceInfo,
+        Statement, StatementKind,
+    },
+    ty::{self, TyCtxt},
+};
+use rustc_span::DUMMY_SP;
+
+use crate::{
+    arg::{ArgKind, InstrumentationArg},
+    deref::remove_outer_deref,
+};
+
+/// Cast an argument from pointer to `usize`, if needed.
+///
+/// Casts `arg` to `usize` if needed, returning the cast statement and new,
+/// `usize`-typed operand if the cast was needed. This cast statement must be
+/// inserted into the function's body before the new operand is used.
+/// `arg` will be used as a copy in the new statement,
+/// so this statement must be inserted in a position where `arg` is alive.
+pub fn cast_ptr_to_usize<'tcx>(
```

Comment:
feels odd to have a module for one function, and I don't anticipate needing to add more

Reply:
I don't think it's that odd.  Would it make more sense in `mir_utils`?

Reply:
i don't think it belongs in `mir_utils` because it references non-mir types, i.e. InstrumentationArg. It should go into `points`, if anything, and some of the internals could be broken up into mir-util-specific behavior and factored out in another PR

Reply:
Can we leave it in it's own module for now and then refactor it into pieces in other modules later?

Reply:
Moved to a submodule of `mod point` in https://github.com/immunant/c2rust/pull/559/commits/8e2e95f0e639c467ca23f1b1a4e19c57fea5c749 and re-exported from `mod point` in https://github.com/immunant/c2rust/pull/559/commits/1982e8d040d9a2b68da5557817793ad373ea2d01.